java/core/src/main/java/com/whylogs/core/DatasetProfile.java

```diff
@@ -0,0 +1,141 @@
+package com.whylogs.core;
+
+import com.whylogs.core.metrics.Metric;
+import com.whylogs.core.schemas.ColumnSchema;
+import com.whylogs.core.schemas.DatasetSchema;
+import com.whylogs.core.views.ColumnProfileView;
+import com.whylogs.core.views.DatasetProfileView;
+import java.time.ZonedDateTime;
+import java.util.*;
+import lombok.Getter;
+import lombok.ToString;
+
+// TODO: extend WRITABLE interface
+@Getter
+@ToString
+public class DatasetProfile {
+  // TODO: Time zone is all mixed up. Fix
+  public static int _LARGE_CACHE_SIZE_LIMIT = 1024 * 100;
+
+  private DatasetSchema schema;
+  // QUESTION: Do we need zones here? Do we just use UTC?
+  private Date datasetTimestamp;
+  private Date creationTimestamp;
+  private HashMap<String, ColumnProfile<?>> columns;
+  private boolean isActive = false;
+  private int trackCount = 0;
+
+  public DatasetProfile(
+      Optional<DatasetSchema> datasetSchema,
+      Optional<Date> datasetaTimestampe,
+      Optional<Date> creationTimestampe) {
+    this.schema = datasetSchema.orElse(new DatasetSchema());
+    this.datasetTimestamp = datasetaTimestampe.orElse(new Date());
+    this.creationTimestamp = creationTimestampe.orElse(new Date());
+
+    this.columns = new HashMap<>();
+    this.initializeNewColumns(schema.getColNames());
+  }
+
+  public void addMetric(String colName, Metric metric) {
+    if (!this.columns.containsKey(colName)) {
+      throw new InputMismatchException("Column name not found in schema");
+    }
+    this.columns.get(colName).addMetric(metric);
+  }
+
+  /*
+  TODO: I don't beleive we need this in Java? (with the T Object)
```

Comment:
@jamie256 I'm not sure we would need this in Java with the T Object as we would just be tracking Objects if it's unknown?  

What are your thoughts?

Reply:
Yeah I don't think we need `T` here.